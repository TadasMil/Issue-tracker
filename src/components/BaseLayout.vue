<script lang="ts" setup name="BaseLayout">
import BaseSidebar from '@/components/BaseSidebar.vue'
</script>

<template>
  <b-container fluid>
    <b-row>
      <b-col class="col-sm-auto p-0">
        <BaseSidebar />
      </b-col>

      <b-col class="col-sm p-3 min-vh-100">
        <slot />
      </b-col>
    </b-row>
  </b-container>
</template>
