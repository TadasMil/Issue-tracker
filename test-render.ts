import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
