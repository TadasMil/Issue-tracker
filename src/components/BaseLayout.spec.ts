import { describe, it, expect, vi } from 'vitest'
import { render } from '../../test-render'

import BaseLayout from './BaseLayout.vue'

vi.mock('vue-router', () => ({
  useRoute: vi.fn(() => ({ path: '/' })),
}))

describe('BaseLayout', () => {
  it('should render the component', () => {
    const { html } = render(BaseLayout, {
      global: {
        stubs: ['BaseSidebar'],
      },
    })

    expect(html()).toMatchSnapshot()
  })

  it('should render the default slot', () => {
    const { html } = render(BaseLayout, {
      slots: {
        default: '<div>Default slot</div>',
      },
      global: {
        stubs: ['BaseSidebar'],
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
