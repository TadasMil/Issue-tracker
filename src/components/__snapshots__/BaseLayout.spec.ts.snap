// Vitest Snapshot v1

exports[`BaseLayout > should render the component 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col col-sm-auto p-0\\">
      <base-sidebar-stub></base-sidebar-stub>
    </div>
    <div class=\\"col col-sm p-3 min-vh-100\\"></div>
  </div>
</div>"
`;

exports[`BaseLayout > should render the default slot 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col col-sm-auto p-0\\">
      <base-sidebar-stub></base-sidebar-stub>
    </div>
    <div class=\\"col col-sm p-3 min-vh-100\\">
      <div>Default slot</div>
    </div>
  </div>
</div>"
`;
